packages/grafana-data/src/types/datasource.ts

```diff
@@ -50,12 +52,16 @@ export class DataSourcePlugin<
     return this;
   }
 
+  /** @deprecated it will be removed in a future release */
   setConfigCtrl(ConfigCtrl: any) {
+    deprecationWarning('DataSourcePlugin', 'setConfigCtrl');
     this.angularConfigCtrl = ConfigCtrl;
     return this;
   }
 
+  /** @deprecated it will be removed in a future release */
   setQueryCtrl(QueryCtrl: any) {
+    deprecationWarning('DataSourcePlugin', 'setQueryCtrl');
```

Comment:
Maybe I am missing something, but is it still going to be a valid scenario - after merging this PR - that a non-angular plugin calls these methods? Isn't "deprecated" a bit misleading in a sense, that these actually won't work after the PR is merged?

If that's case, would it make sense to remove the methods / throw an error?